test/test_dispatch.py

```diff
@@ -571,6 +571,44 @@ def test_computed_table_with_ambiguous_autogradother(self):
 AutogradCPU: fn_math [math kernel]
 AutogradCUDA: fn_math [math kernel]
 AutogradXLA: fn_math [math kernel]
+''')
+
+    def test_computed_table_with_cpu_autograd_math_defaultbackend(self):
+        result = self.commute("foo", [
+            # m.def("foo(Tensor x) -> Tensor")
+            lambda m: m.def_("foo(Tensor x) -> Tensor"),
+            # m.impl("foo", torch::kCPU, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "CPU", debug="fn_cpu"),
+            # m.impl("foo", torch::kAutograd, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "Autograd", debug="fn_autograd"),
+            # m.impl("foo", torch::kMath, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "Math", debug="fn_math"),
+            # m.impl("foo", torch::kDefaultBackend, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "DefaultBackend", debug="fn_defaultbackend"),
+        ])
+        state, table = result.state, result.table
+        self.assertExpectedInline(state, '''\
+name: test::foo
+schema: test::foo(Tensor x) -> (Tensor)
+debug: registered at /dev/null:0
+alias analysis kind: FROM_SCHEMA
+CPU: fn_cpu :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+Autograd[alias]: fn_autograd :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+Math[alias]: fn_math :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+DefaultBackend[alias]: fn_defaultbackend :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+''')
+
+        # computed dispatch table is too big, so we only check on a few entries we're interested in.
+        extracted_table = extract_dispatch_table_with_keys(table, dispatch_keys_to_check)
+
+        self.assertExpectedInline(extracted_table, '''\
+CPU: fn_cpu [kernel]
+CUDA: fn_defaultbackend [default backend kernel]
+XLA: fn_defaultbackend [default backend kernel]
+AutogradOther: fn_math [math kernel]
+AutogradCPU: fn_autograd [autograd kernel]
+AutogradCUDA: fn_math [math kernel]
+AutogradXLA: fn_math [math kernel]
```

Comment:
This feels... wrong? Default backend should count as a backend for the purposes of math registration, which means you should get `fn_autograd` here? (So it seems to me like Math shouldn't show up at all)